test/e2e/resources/nlb_stack.yaml

```diff
@@ -0,0 +1,239 @@
+Parameters:
+  EnvironmentName:
+    Description: An environment name that is prefixed to resource names
+    Type: String
+
+  VpcCIDR:
+    Description: Please enter the IP range (CIDR notation) for this VPC
+    Type: String
+    Default: 10.192.0.0/16
+
+  PublicSubnet1CIDR:
+    Description: Please enter the IP range (CIDR notation) for the public subnet in the first Availability Zone
+    Type: String
+    Default: 10.192.10.0/24
+
+  PublicSubnet2CIDR:
+    Description: Please enter the IP range (CIDR notation) for the public subnet in the second Availability Zone
+    Type: String
+    Default: 10.192.11.0/24
+
+  PrivateSubnet1CIDR:
+    Description: Please enter the IP range (CIDR notation) for the private subnet in the first Availability Zone
+    Type: String
+    Default: 10.192.20.0/24
+
+  PrivateSubnet2CIDR:
+    Description: Please enter the IP range (CIDR notation) for the private subnet in the second Availability Zone
+    Type: String
+    Default: 10.192.21.0/24
+
+Resources:
```

Comment:
We have never used CF for ack testing before, and i'm little hesitant here. In [acktest](https://github.com/aws-controllers-k8s/test-infra/blob/main/src/acktest/bootstrapping/vpc.py) we have a library that helps pre-provisioning resources for testing, and it has already enough for VPC/Subnets/Routes/NatGateways - maybe we can consider add a little to support NLB's?

Reply:
CloudFormation is commonly used for provisioning resources that'd otherwise require a lot of code for creating and deleting resources after establishing dependencies among them. From a cursory look at `acktest/bootstrapping`, it looks like it either creates public subnets or fully-private subnets. Our tests use private subnets that route to a NAT gateway. For this case, I'm in favor of using CloudFormation over `acktest` as `acktest` would require additional effort to support NLBs and the aforementioned type of private subnets. 

Also, since our future plans are to move away from using Python for testing, using CloudFormation will help get us closer to that goal as migration will only require making an API call with the same template. WDYT?

Reply:
I've been mulling this over, and to be honest, I'm not sold on the idea of using CF for our tests. It feels like we might be taking a step in the wrong direction, not just for our project but for the bigger picture of what we're trying to achieve with ACK. I get why it might seem appealing at first glance, but I've got a few thoutghts:

- We' re actually in the process of moving our test infrastructure from CDK to ACK. It seems counterintuitive to introduce CF now when we're actively working to phase it out in other areas
- Throwing CF into the mix might make our codebase a bit of a maze. We've successfully used `acktest/bootstrap` for 44 controllers already, and it's been working quite well for us. Introducing a new approach now could throw off folks trying to contribute or understand our setup..
- To wrap up, I've had my fair share of headaches with CF - it can be painfully slow, and stuck rollbacks are no joke. FYI, between PR #7 and PR #8, tests went from 6m22s to 20m9s, and I don't want to imagine where this is heading. Keeping all 44 controllers patching below two hours is at risk if we take this approach

Reply:
>From a cursory look at acktest/bootstrapping, it looks like it either creates public subnets or fully-private subnets.

Bootstrap library can create both public and private subnets, you can see an example of this in the eks controller: https://github.com/aws-controllers-k8s/eks-controller/blob/main/test/e2e/service_bootstrap.py#L56

> Our tests use private subnets that route to a NAT gateway. 

his is configurable and doesn't necessarily depend on how our prow infrastructure is set up. we're creating these resources in a different accounts, so we have the freedom to configure them as we wish.

Reply:
>Also, since our future plans are to move away from using Python for testing, using CloudFormation will help get us closer to that goal as migration will only require making an API call with the same template.

our plans to move from Python to (generated) Go for testing didn't factor in any CF usage. The idea is to reuse ACK controllers and similar bootstrapping mechanisms to what we currently do in Python (but generated).

Reply:
> between PR https://github.com/aws-controllers-k8s/apigateway-controller/pull/7 and PR https://github.com/aws-controllers-k8s/apigateway-controller/pull/8, tests went from 6m22s to 20m9s, and I don't want to imagine where this is heading

The increase in test runtime is largely related to the VPC Link tests creating an entire VPC networking stack, an ELB, and waiting for the `VPCLink` resource to transition to the `AVAILABLE` state (which alone takes a few minutes). 

> To wrap up, I've had my fair share of headaches with CF - it can be painfully slow, and stuck rollbacks are no joke

Stuck rollbacks or issues with deletions usually happen if a resource created outside of CloudFormation has a dependency on a resource provisioned via CloudFormation. `acktest/bootstrap` can also suffer from deletion issues if a test creates a resource dependent on an `acktest/bootstrap` resource. 


> his is configurable and doesn't necessarily depend on how our prow infrastructure is set up. we're creating these resources in a different accounts, so we have the freedom to configure them as we wish.

I more meant that this has the added effort of implementing it in `acktest/bootstrap`. 

> Throwing CF into the mix might make our codebase a bit of a maze. Introducing a new approach now could throw off folks trying to contribute or understand our setup.

Good point. I do agree that being consistent is reason enough to amend `acktest/bootstrap` to support the missing resources, in lieu of using CloudFormation. I'll remove usage of CloudFormation in favor of adding the missing resources to `acktest/bootstrap`.